src/support/threads.cpp

```diff
@@ -111,7 +111,14 @@ void ThreadPool::initialize(size_t num) {
   ready.store(threads.size()); // initial state before first resetThreadsAreReady()
   resetThreadsAreReady();
   for (size_t i = 0; i < num; i++) {
-    threads.emplace_back(make_unique<Thread>());
+    try {
```

Comment:
Nit: should the `try/catch` surround the `for` loop, instead of the other way around? They probably get optimized identically so it's probably totally moot.

Reply:
I usually prefer to put the try-catch as close to the possibly-throwing code as possible, but maybe that's more of a dynamic language thing... anyhow, yeah, I doubt it matters here in practice.